src/components/NewWorkspaceModal.js

```diff
@@ -83,6 +83,14 @@ const NewWorkspaceModal = withDisplayName('NewWorkspaceModal', ({
     setIsAlphaRegionalityUser(await Ajax(signal).Groups.group(getConfig().alphaRegionalityGroup).isMember())
   })
 
+  // Error is ignored here since any errors associated with Leo app creation are not communicated during workspace creation
+  // Rather when the user first visits the `Data` tab
+  const createLeoApp = withErrorIgnoring(async workspace => {
+    if (isAzureBillingProject() && !getConfig().isProd) {
+      await Ajax().Apps.createAppV2(`wds-${workspace.workspaceId}`, workspace.workspaceId)
+    }
+  })
```

Comment:
I _think_ that you could remove the async/await here:
```javascript
  const createLeoApp = withErrorIgnoring(workspace => {
    if (isAzureBillingProject() && !getConfig().isProd) {
      Ajax().Apps.createAppV2(`wds-${workspace.workspaceId}`, workspace.workspaceId)
    }
  })
```
BUT, since this ignores errors and nothing relies on the result of `createLeoApp`, the change to remove async/await is invisible to users; it just makes slightly more work for the JS engine. I'm fine leaving this as-is.

Reply:
We aren't awaiting `createLeoApp`, so the creation of the workspace proceeds before this finishes. But the async/await here is necessary to make sure that the error handling happens.

(See https://github.com/DataBiosphere/terra-ui/pull/3699#discussion_r1081889331)

Reply:
ah - `withErrorIgnoring` needs the async/await, that makes sense, thank you I didn't realize that!